### PR TITLE
fix for deleting multiple directory message 47892

### PR DESCRIPTION
**Pull request by pradeepmurugesan**
#47892

### CODE REVIEW

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts [hunk 1]
```diff
@@ -633,9 +633,8 @@ class BaseDeleteFileAction extends BaseFileAction {
 
 			// Confirm for moving to trash
 			else if (this.useTrash) {
-				const message = distinctElements.length > 1 ? getConfirmMessage(nls.localize('confirmMoveTrashMessageMultiple', "Are you sure you want to delete the following {0} files?", distinctElements.length), distinctElements.map(e => e.resource))
-					: distinctElements[0].isDirectory ? nls.localize('confirmMoveTrashMessageFolder', "Are you sure you want to delete '{0}' and its contents?", distinctElements[0].name)
-						: nls.localize('confirmMoveTrashMessageFile', "Are you sure you want to delete '{0}'?", distinctElements[0].name);
+				const message = this.getMoveToTrashMessage(distinctElements);
```
Comment (bpasero) [inline src/vs/workbench/parts/files/electron-browser/fileActions.ts]: @pradeepmurugesan it seems to me the same issue exists for the code below (after the comment "Confirm for deleting permanently") when bypassing the trash. Can we enrich your existing method `getMoveToTrashMessage` to also handle the case of deleting without trash? Then maybe it should be one method `getDeleteMessage` that has a parameter for wether the trash is used or not.

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts [hunk 2]
```diff
@@ -724,6 +722,43 @@ class BaseDeleteFileAction extends BaseFileAction {
...
 	}
+
+	private getDeleteMessage(distinctElements: Array<ExplorerItem>, deletePermanentaly: boolean = false): string {
+		const key = deletePermanentaly ? 'DeleteMessage' : 'MoveTrash';
+		const messages = this.getMessages(deletePermanentaly);
+		let message: string = '';
+
+		if (distinctElements.length > 1) {
+			if (distinctElements[0].isDirectory) {
+				message = getConfirmMessage(nls.localize(`confirm${key}MessageMultipleDirectories`, messages[`confirm${key}MessageMultipleDirectories`], distinctElements.length), distinctElements.map(e => e.resource));
```
Comment (pradeepmurugesan) [inline src/vs/workbench/parts/files/electron-browser/fileActions.ts]: @bpasero I am not sure about interpolating a variable here. But I couldn't find any other easier way to remove the duplicate code, if used the keys as it is..

**Review by bpasero — CHANGES_REQUESTED** [review]
@pradeepmurugesan even if it means more duplication, for our nls story we cannot have dynamic keys. So we need to stick with individual `nls.localize()` calls that have the string within. And I would not start to introduce a dictionary or any other helper.

**Comment by pradeepmurugesan** [comment]
@bpasero In that case why can't we just have 2 methods separately to fetch the messages.. Having a flag and duplicating the if else logic is going to make the method huge and I am not seeing the advantage of having the flag.

kindly let me know if I am missing something

**Comment by bpasero** [comment]
@pradeepmurugesan sure 2 methods is also fine

**Comment by pradeepmurugesan** [comment]
@bpasero Great thanks.. have updated.. Kindly review

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts [hunk 3]
```diff
@@ -725,6 +722,42 @@ class BaseDeleteFileAction extends BaseFileAction {
 			});
 		});
 	}
+
+	private getMoveToTrashMessage(distinctElements): string {
```
Comment (bpasero) [inline src/vs/workbench/parts/files/electron-browser/fileActions.ts]: @pradeepmurugesan the `distinctElements` is missing a type annotation: `distinctElements: ExplorerItem[]`

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts [hunk 4]
```diff
@@ -725,6 +722,42 @@ class BaseDeleteFileAction extends BaseFileAction {
...
+			if (distinctElements[0].isDirectory) {
+				message = nls.localize('confirmMoveTrashMessageFolder', "Are you sure you want to delete '{0}' and its contents?", distinctElements[0].name);
+			} else {
+				message = nls.localize('confirmMoveTrashMessageFile', "Are you sure you want to delete '{0}'?", distinctElements[0].name);
+			}
+		}
+		return message;
+	}
+
+	private getDeleteMessage(distinctElements): string {
```
Comment (bpasero) [inline src/vs/workbench/parts/files/electron-browser/fileActions.ts]: @pradeepmurugesan the `distinctElements` is missing a type annotation: `distinctElements: ExplorerItem[]`

**Review by bpasero — CHANGES_REQUESTED** [review]
@pradeepmurugesan there is still an issue when the selection of files and folders is mixed, e.g. select a folder and a file and hit delete, the message wrongly indicates to me that the 2 things are both folders. I fear we have to distinguish that case as well or just talk about files/folders in the case of a multi selection if we do not want to be more specific.

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts (hunk 3 above)
Comment (pradeepmurugesan) [reply src/vs/workbench/parts/files/electron-browser/fileActions.ts]: fixed

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts (hunk 4 above)
Comment (pradeepmurugesan) [reply src/vs/workbench/parts/files/electron-browser/fileActions.ts]: fixed

**Comment by pradeepmurugesan** [comment]
@bpasero I didn't think about the Hybrid case earlier.. Fixed the same now.  Kindly review

**Comment by bpasero** [comment]
@isidorn if you want to merge for April, feel free to review. otherwise running out of time, moving to May.

**Comment by isidorn** [comment]
@bpasero May is good

**Pull request merged**